examples/progress.py

```diff
@@ -0,0 +1,61 @@
+import random
+from time import sleep
+
+from magicgui import magicgui
+from magicgui.tqdm import tmgrange, tqdm_mgui
+from magicgui.widgets import ProgressBar
+
+# can be used alone (outside of a magicgui)
+# `tmgrange(*args, **kwargs)` is a shortcut for tqdm_mgui(range(*args), **kwargs)
+# (..., leave=False) will make the progressbar hide when done
+for i in tmgrange(10, leave=False):
+    sleep(0.05)
+
+
+# can also be used inside of a magicgui decorator
+# in which case the progress bar will be added to the magicgui container
+@magicgui(call_button=True)
+def long_running(steps=10, delay=0.1):
```

Comment:
If functions with `tmgrange` or `tqdm_mgui` inside then don't get magic gui decorated will they just behave as normal with the default behavior of of `tqdm`?

Reply:
at the moment no, they will also pop up a little GUI progress bar (it just won't be in any parent widget)... but that is absolutely possible and definitely a goal (i.e. it should be smart enough to detect the context in which it is called)

Reply:
fixed... now it will just be the default behavior

Reply:
nice! i think that's pretty cool!!! enables maximum reusability